### PR TITLE
update z3 in docker files

### PR DESCRIPTION
**Pull request by leonardoalt**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -75,7 +75,7 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
-set(LATEST_Z3_VERSION "4.11.2")
+set(LATEST_Z3_VERSION "4.12.1")
```
Comment (ekpyron) [inline CMakeLists.txt]: We should really only do this in the second PR we create, s.t. builds don't break here, shouldn't we :-)? Or did you want to squeeze everything into one PR by constant rebasing and cancelling later docker build runs :-)?

Comment (leonardoalt) [reply CMakeLists.txt]: Yea we shouldn't merge that here, but without this the build won't go through inside docker

Comment (ekpyron) [reply CMakeLists.txt]: Ah, yeah, we broke the flow with the strict version requirement... I guess we should disable the strict Z3 version requirement in the docker builds...

**Comment by leonardoalt** [comment]
@ekpyron next one

**Review by ekpyron — DISMISSED** [review]

**Comment by ekpyron** [comment]
We should also probably flip this
```
      # Build without Z3. We won't be running SMT tests on Arch because that requires a specific
      # version of Z3 and the one with the official repos is often not the one we need.
      USE_Z3: OFF
```
to ``ON`` again in ``.circleci/config.yml`` and instead note in the comment that it can be switched off, if we can't keep up for some reason - that way we're autoreminded of this...

**Comment by leonardoalt** [comment]
@ekpyron fixed the above. Merge?

**Review by ekpyron — APPROVED** [review]

**Pull request merged**